src/backend/common/helpers/alliance_helper.py

```diff
@@ -0,0 +1,181 @@
+from typing import Dict, List, Optional, Tuple
+
+from pyre_extensions import none_throws
+
+from backend.common.consts.comp_level import COMP_LEVELS_VERBOSE_FULL, CompLevel
+from backend.common.consts.event_type import EventType
+from backend.common.consts.playoff_type import DoubleElimRound, PlayoffType
+from backend.common.models.alliance import (
+    EventAlliance,
+    PlayoffAllianceStatus,
+    PlayoffOutcome,
+)
+from backend.common.models.event import Event
+from backend.common.models.keys import TeamKey
+
+
+class AllianceHelper:
+    UNKNOWN_ALLIANCE_SIZE = 99
+
+    KNOWN_MODERN_EVENT_TYPE_ALLIANCE_SIZES: Dict[EventType, int] = {
+        EventType.REGIONAL: 3,
+        EventType.DISTRICT: 3,
+        EventType.DISTRICT_CMP: 3,
+        EventType.DISTRICT_CMP_DIVISION: 3,
+        EventType.CMP_DIVISION: 4,
+        EventType.CMP_FINALS: 4,
+    }
+
+    @classmethod
+    def get_known_alliance_size(
+        cls,
+        event_type: EventType,
+        year: int,
+    ) -> int:
+
+        if year < 2014:
+            # 2014 was first year of 4 team champs divisions
+            # for now, return default for legacy events
+            return cls.UNKNOWN_ALLIANCE_SIZE
+
+        return cls.KNOWN_MODERN_EVENT_TYPE_ALLIANCE_SIZES.get(
+            event_type, cls.UNKNOWN_ALLIANCE_SIZE
+        )
+
+    @staticmethod
+    def get_ordinal_pick_from_number(n: int) -> str:
+        # https://stackoverflow.com/a/20007730
+        if 11 <= (n % 100) <= 13:
+            suffix = "th"
+        else:
+            suffix = ["th", "st", "nd", "rd", "th"][min(n % 10, 4)]
+        return f"{n}{suffix} Pick"
+
+    @classmethod
+    def get_alliance_details_and_pick_name(
+        cls, event: Event, team: TeamKey
+    ) -> Tuple[Optional[EventAlliance], Optional[str]]:
+        alliance = None
+        alliance_pick = None
+
+        if event.alliance_selections is None:
+            return (None, None)
+
+        alliance_size = cls.get_known_alliance_size(event.event_type_enum, event.year)
+
+        for alliance_check in none_throws(event.alliance_selections):
```

Comment:
if either of the `if` statements below match, we should be able to break out of the for loop right?